llvm/test/CodeGen/AMDGPU/llvm.amdgcn.workitem.id-unsupported-calling-convention.ll

```diff
@@ -0,0 +1,61 @@
+; NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=amdgcn -mcpu=gfx942 -O0 -stop-after=si-fix-sgpr-copies -o - %s | FileCheck --check-prefix=SelDAG %s
+; RUN: llc -global-isel -mtriple=amdgcn -mcpu=gfx942 -stop-after=legalizer -o - %s | FileCheck --check-prefix=GlobalISel %s
+
+declare i32 @llvm.amdgcn.workitem.id.x()
+declare i32 @llvm.amdgcn.workitem.id.y()
+declare i32 @llvm.amdgcn.workitem.id.z()
+
+define amdgpu_ps void @undefined_workitems(ptr %p, ptr %q, ptr %r) {
```

Comment:
I don't see this function is marked with `amdgpu-no-*` attribute. Is it implied by the CC?

Reply:
I assumed so since:
- GlobalISel behaves this way, too.
- the [docs for “amdgpu-no-workitem-id-x”](https://llvm.org/docs/AMDGPUUsage.html#amdgpu-llvm-ir-attributes-table) say: "Indicates the function does not depend on the value of the llvm.amdgcn.workitem.id.x intrinsic. [...] The backend can generally infer this during code generation, so typically there is no benefit to frontends marking functions with this."
- the (spare) [docs on CC](https://llvm.org/docs/AMDGPUUsage.html#amdgpu-cc) specify for `amdgpu_ps`, for example: "Used for Mesa/AMDPAL pixel shaders. ..TODO:: Describe." which led me to assume that it does not guarantee the (Initial Kernel Execution State)[https://llvm.org/docs/AMDGPUUsage.html#initial-kernel-execution-state], as described for `amdgpu_kernel`.